### PR TITLE
Fix #914 Unable to set Vanity URL

### LINKED ISSUE #914: Shared Links: Unable to set Vanity URL
- [ X] I have checked that the [SDK documentation][sdk-docs] doesn't solve my issue.
- [ X] I have checked that the [API documentation][api-docs] doesn't solve my issue.
- [ X] I have searched the [Box Developer Forums][dev-forums] and my issue isn't already reported (or if it has been reported, I have attached a link to it, for reference).
- [ X] I have searched [Issues in this repo][github-repo] and my issue isn't already reported.

### Description of the Issue
The Java SDK does not allow you to set a vanity URL for a shared link. When you get the shared link information the vanity url will be included but there is not method for setting it. 

### Steps to Reproduce
Execute the following CURL to set:
`curl --location --request PUT 'https://api.box.com/2.0/folders/<folder_id>?fields=shared_link' \
--header 'Content-Type: application/json' \

--data-raw '{
    "shared_link": {
        "access": "open",
        "password": "do-not-use-this-password",

        "vanity_url": "https://<domain>.box.com/v/abcdefghijklmno",
        "vanity_name": "abcdefghijklmno",

        "permissions": {
            "can_download": true
        }
    }
}'`

### Expected Behavior
Shared link is created with a vanity url

### Error Message, Including Stack Trace
NA

### Versions Used
Java SDK: 2.55
Java: AdoptJDK 11.0.11

[sdk-docs]: ./doc
[api-docs]: https://developer.box.com/docs
[dev-forums]: https://community.box.com/t5/Platform-and-Development-Forum/bd-p/DeveloperForum
[github-repo]: https://github.com/box/box-java-sdk/search?type=Issues

### PR DESCRIPTION
**Pull request by antusus**
We cannot set Vanity URL directly but we can set Vanity Name. This name is later used in generated link.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,7 @@ __Breaking Changes:__
 __New Features and Enhancements:__
 
 __Bug Fixes:__
+ - Fix unable to set Vanity URL on `BoxSharedLink` ([#914](https://github.com/box/box-java-sdk/issues/914))
```
Comment (PJSimon) [inline CHANGELOG.md:10]:
```suggestion
 - Fix unable to set Vanity URL on `BoxSharedLink` ([#921](https://github.com/box/box-java-sdk/issues/921))
```

Comment (PJSimon) [reply CHANGELOG.md:10]: The link should point to this PR.

File: src/main/java/com/box/sdk/BoxWebLink.java [hunk 2]
```diff
@@ -49,11 +50,7 @@ public BoxSharedLink createSharedLink(BoxSharedLink.Access access, Date unshareD
                                           BoxSharedLink.Permissions permissions) {
 
         BoxSharedLink sharedLink = new BoxSharedLink(access, unshareDate, permissions);
-        Info info = new Info();
-        info.setSharedLink(sharedLink);
-
-        this.updateInfo(info);
-        return info.getSharedLink();
+        return this.createSharedLink(sharedLink);
```
Comment (PJSimon) [inline src/main/java/com/box/sdk/BoxWebLink.java:53]:
```suggestion
        return this.createSharedLink(new BoxSharedLink(access, unshareDate, permissions));
```

Comment (PJSimon) [reply src/main/java/com/box/sdk/BoxWebLink.java:53]: Just for consistency with BoxFile and BoxFolder.

File: src/main/java/com/box/sdk/BoxWebLink.java [hunk 3]
```diff
@@ -69,13 +66,18 @@ public BoxSharedLink createSharedLink(BoxSharedLink.Access access, Date unshareD
                                           BoxSharedLink.Permissions permissions, String password) {
 
         BoxSharedLink sharedLink = new BoxSharedLink(access, unshareDate, permissions, password);
-        Info info = new Info();
+        return this.createSharedLink(sharedLink);
```
Comment (PJSimon) [inline src/main/java/com/box/sdk/BoxWebLink.java:69]:
```suggestion
        return this.createSharedLink(new BoxSharedLink(access, unshareDate, permissions, password));
```

Comment (PJSimon) [reply src/main/java/com/box/sdk/BoxWebLink.java:69]: Just for consistency with BoxFile and BoxFolder.

**Comment by antusus** [comment]
Converting to draft as we still want to deal with problem of creating SharedLinks on WebLink. Currently when you create a shared link with permissions on a WebLink it will fail. We have to prepare better solution.

**Pull request closed without merging**